resources/models/reservation.py

```diff
@@ -364,6 +364,21 @@ def clean(self, **kwargs):
             if day and not is_valid_time_slot(dt, self.resource.slot_size, day['opens']):
                 raise ValidationError(_("Begin and end time must match time slots"), code='invalid_time_slot')
 
+        # Check if Unit has disallow_overlapping_reservations value of True
+        if (
+            self.resource.unit.disallow_overlapping_reservations and not
+            self.resource.can_create_overlapping_reservations(user)
+        ):
+            reservations_for_same_unit = Reservation.objects.filter(user=user, resource__unit=self.resource.unit)
```

Comment:
We should probably check if overlapping reservations are active (not cancelled). We probably want to allow a situation where users old reservation is cancelled and he then makes a new reservation for that time period.

Reply:
Good catch. Changed!